.github/workflows/ci.yml

```diff
@@ -12,17 +12,17 @@ jobs:
       matrix:
         javaversion: ["8", "9", "10", "11", "12", "13", "14", "15", "16", "17"]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Set up Java ${{ matrix.javaversion }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
```

Comment:
v2 added more requirements:

![Screen Shot 2022-03-25 at 11 51 05 AM](https://user-images.githubusercontent.com/39606064/160175041-c1511ef0-1638-49cd-9ac4-e3cc03b8e950.png)

See their docs on how to fix this: https://github.com/actions/setup-java

Reply:
I am using `v3` as it was released a month ago.

Reply:
I understand, but v3 contains everything that v2 does and we are upgrading from v1 which means you are picking up all the required changes from v2. The build failed because the now required `distribution` is missing from the GitHub Action.

Reply:
I have update the `setup-java@v1` to `v2` and added distribution `zulu`.